### DISCUSSION TITLE
Freedom to shape the logical view of the project.

### DISCUSSION THREAD

**Discussion by DKnoto** [albilu/netbeansPython#93] [Ideas]
[Continuation of submission #90](https://github.com/albilu/netbeansPython/issues/90)

> > 1. In section Sources directory "src" should not be visible just as in the "Tests" section we do not see the "tests" directory;
> 
> Its the expected behaviour. "Sources": is the logical view for all files under the root project folder exept files in "tests" and some other priviledged files.
> 
> There is 2 main python project structure https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/
> 
> If you want the flat layout. Just delete the src folder and put your files directly under the root folder
> 

Here I think we misunderstood a bit. I, for one, would like to have the freedom to determine the logical view of the project regardless of the physical layout of the files on disk, which, as you wrote, should follow Python guidelines.

I wish there was as much freedom as there is for C/C++ projects:

![No-90-Both-Logical-Physical-C-Project-c255](https://github.com/albilu/netbeansPython/assets/1998807/f5805af0-38f2-4b08-a87d-18adf4c0366d)

The logical view does not show everything that is physically in the directory regardless of how the IDE's global filters are set.

The structure of the logical view is not closely related to the physical structure and can be freely shaped.

**Comment by albilu** [comment]
> The structure of the logical view is not closely related to the physical structure and can be freely shaped.

Aggreed. I made that choice to be flexible considering the src layout and flat layout and that most python project uses the flat layout.

I am not sure if there is a freedom to shape the C++ project logical view. These choices are hardcoded for example if i dont want to have the header files i am not sure to have the choice
